tests/Browser/Pages/Backend/Posts/PostDetailTest.php

```diff
@@ -0,0 +1,407 @@
+                    ->assertVisible('.post-image[src="' . $post->image_url . '"]');
+            $this->assertTrue($browser->text('.post-username') == $post->users->name);
+            $this->assertTrue($browser->text('.post-date') == date('d-m-Y', strtotime($post->created_at)));
+            $this->assertTrue($browser->text('.post-content') == $post->content);
+        });
+    }
+
+    /**
+     * Detail post is status
+     *
+     * @return void
+     */
+    public function testClickLinkTransfertToDetaiIsStatuslHaveComment()
+    {
+        $this->makeDataStatus(2);
+        $post = Post::first();
+        $this->browse(function (Browser $browser) use ($post) {
+            $browser->loginAs($this->user)
+                    ->resize(1200, 1600)
+                    ->visit('/admin/posts/' . $post->id)
+                    ->assertPathIs('/admin/posts/' . $post->id)
+                    ->assertSee('Detail Post')
+                    ->assertSee('Date')
+                    ->assertSee('STATUS')
+                    ->assertDontSee('Score')
+                    ->assertVisible('.post-image[src="' . $post->image_url . '"]');
+            $this->assertTrue($browser->text('.post-username') == $post->users->name);
+            $this->assertTrue($browser->text('.post-date') == date('d-m-Y', strtotime($post->created_at)));
+            $this->assertTrue($browser->text('.post-content') == $post->content);
+        });
+    }
+
+    /**
+     * Detail post is status
+     *
+     * @return void
+     */
+    public function testClickLinkTransfertToDetaiIsFindBooklHaveComment()
+    {
+        $this->makeDataFindBook(2);
+        $post = Post::first();
+        $this->browse(function (Browser $browser) use ($post) {
+            $browser->loginAs($this->user)
+                    ->resize(1200, 1600)
+                    ->visit('/admin/posts/' . $post->id)
+                    ->assertPathIs('/admin/posts/' . $post->id)
+                    ->assertSee('Detail Post')
+                    ->assertSee('Date')
+                    ->assertSee('FIND BOOK')
+                    ->assertDontSee('Score')
+                    ->assertVisible('.post-image[src="' . $post->image_url . '"]');
+            $this->assertTrue($browser->text('.post-username') == $post->users->name);
+            $this->assertTrue($browser->text('.post-date') == date('d-m-Y', strtotime($post->created_at)));
+            $this->assertTrue($browser->text('.post-content') == $post->content);
+        });
+    }
+
+
+    /**
+     * Make data has comment
+     *
+     * @return void
+     */
+    public function makeData($row)
+    {
+        $faker = Faker::create();
+
+        $category = factory(Category::class)->create();
+        $donator = factory(Donator::class)->create([
+            'user_id' => $this->user->id,
+        ]);
+
+        $book = factory(Book::class)->create([
+            'category_id' => $category->id,
+            'donator_id' => $donator->id,
+        ]);
```

Comment:
$faker = Faker::create();
 +
 +        $category = factory(Category::class)->create();
 +        $donator = factory(Donator::class)->create([
 +            'user_id' => $this->user->id,
 +        ]);
 +
 +        $book = factory(Book::class)->create([
 +            'category_id' => $category->id,
 +            'donator_id' => $donator->id,
 +        ]);
đưa cái này vô 1 hàm Luân.
rồi return vê []
thấy hàm nào cũng lặp đoạn này hết.